client/src/components/RichTextEditor.js

```diff
@@ -372,40 +450,18 @@ const Element = ({ attributes, children, element }) => {
       return <li {...attributes}>{children}</li>
     case "block-quote":
       return <blockquote {...attributes}>{children}</blockquote>
-    case ANET_LINK:
-    case EXTERNAL_LINK: {
-      const reducedChildren = element.children.reduce(
-        (acc, child) => acc + child.text,
-        ""
-      )
-      return (
-        <span
-          {...attributes}
-          style={{
-            padding: "1px",
-            verticalAlign: "baseline",
-            display: "inline-block",
-            borderRadius: "4px",
-            boxShadow: selected && focused ? "0 0 0 2px #B4D5FF" : "none"
-          }}
-        >
-          {element.type === ANET_LINK ? (
-            <LinkAnetEntity
-              type={element.entityType}
-              uuid={element.entityUuid}
-              displayCallback={displayCallback}
-            >
-              {reducedChildren}
-            </LinkAnetEntity>
-          ) : (
-            <LinkExternalHref url={element.url}>
-              {reducedChildren}
-            </LinkExternalHref>
-          )}
-          {children}
-        </span>
-      )
+    case "link": {
+      const entityInfo = getEntityInfoFromUrl(element.url)
+      if (entityInfo.type === ANET_LINK) {
+        return getAnetLink(entityInfo, children, attributes, selected, focused)
+      } else {
+        return getExternalLink(element, children, attributes, selected, focused)
+      }
     }
+    case ANET_LINK:
+      return getAnetLink(element, children, attributes, selected, focused)
+    case EXTERNAL_LINK:
+      return getExternalLink(element, children, attributes, selected, focused)
```

Comment:
I'd change these cases to:
```
    case "link":
    case ANET_LINK:
    case EXTERNAL_LINK:
      return getLink(element, children, attributes, selected, focused)
```
with `getLink` defined as:
```
const getLink = (element, children, attributes, selected, focused) => {
  const reducedChildren = element.children.reduce(
    (acc, child) => acc + child.text,
    ""
  )
  const entityInfo = getEntityInfoFromUrl(element.url)
  const linkElement =
    entityInfo.type === ANET_LINK ? (
      <LinkAnetEntity
        type={entityInfo.entityType}
        uuid={entityInfo.entityUuid}
        displayCallback={displayCallback}
      >
        {reducedChildren}
      </LinkAnetEntity>
    ) : (
      <LinkExternalHref url={entityInfo.url} attributes={attributes}>
        {reducedChildren}
      </LinkExternalHref>
    )

  return (
    <span
      {...attributes}
      style={{
        padding: "1px",
        verticalAlign: "baseline",
        display: "inline-block",
        borderRadius: "4px",
        boxShadow: selected && focused ? "0 0 0 2px #B4D5FF" : "none"
      }}
    >
      {linkElement}
      {children}
    </span>
  )
}
```
(and then you can remove `getExternalLink` and `getAnetLink`).